### DISCUSSION TITLE
Issue timeline quotes response with double quotes

### DISCUSSION THREAD

**Discussion by k0shk0sh** [community/community#24477] [Apps, API and Webhooks]
<p>Hello there, </p>
<p>I’m the author of FastHub app and i’m trying to migrate my app fully to Graphql however there are some APIs haven’t been yet to migrate but I lived with that however i’m facing a wall right now as i’m trying to render issue timeline but end up with <code>""</code> response trying to receive full timeline response, please refer to the example below: </p>
<pre><code>{
  user(login: "k0shk0sh") {
    repository(name: "FastHub") {
      issue(number: 2224) {
        timeline(first: 30) {
          totalCount
          nodes {
            __typename
            ... on Commit {
              author {
                name
                avatarUrl
              }
              message
            }
            ... on IssueComment {
              author {
                login
                avatarUrl
              }
              createdAt
              authorAssociation
              bodyHTML
              body
              id
              updatedAt
              viewerCanReact
              viewerCanDelete
              viewerCanUpdate
              viewerDidAuthor
              viewerCanMinimize
              viewerCannotUpdateReasons
              reactionGroups {
                content
                createdAt
                viewerHasReacted
              }
            }
            ... on CrossReferencedEvent {
              createdAt
              isCrossRepository
              referencedAt
              source {
                __typename
              }
              willCloseTarget
              actor {
                login
                avatarUrl
              }
            }
            ... on ClosedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              closer {
                ... on Commit {
                  abbreviatedOid
                  author {
                    name
                    avatarUrl
                  }
                  message
                }
                ... on PullRequest {
                  number
                  author {
                    avatarUrl
                    login
                  }
                  merged
                }
              }
            }
            ... on ReopenedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
            }
            ... on SubscribedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
            }
            ... on UnsubscribedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
            }
            ... on ReferencedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              commitRepository {
                nameWithOwner
              }
              isCrossRepository
              isDirectReference
              commit {
                message
                committedDate
                abbreviatedOid
              }
              subject {
                ... on Issue {
                  number
                  author {
                    login
                    avatarUrl
                  }
                  repository {
                    nameWithOwner
                  }
                }
                ... on PullRequest {
                  number
                  author {
                    login
                    avatarUrl
                  }
                  repository {
                    nameWithOwner
                  }
                }
              }
            }
            ... on AssignedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
              user {
                login
                avatarUrl
              }
            }
            ... on UnassignedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
              user {
                login
                avatarUrl
              }
            }
            ... on LabeledEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              label {
                color
                name
                isDefault
              }
            }
            ... on UnlabeledEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              label {
                color
                name
                isDefault
              }
            }
            ... on MilestonedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              milestoneTitle
            }
            ... on DemilestonedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              milestoneTitle
            }
            ... on RenamedTitleEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              currentTitle
              previousTitle
            }
            ... on LockedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
              lockReason
              lockable {
                activeLockReason
              }
            }
            ... on UnlockedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
              lockable {
                activeLockReason
              }
            }
            ... on TransferredEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              fromRepository {
                id
              }
              issue {
                number
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>running above in the explorer always result in “” reponse with empty data. </p>
<p>However if I truncate the request to below it returns data just fine which isn’t something I want but rather I would like to receive the full data of an issue. </p>
<pre><code>{
  user(login: "k0shk0sh") {
    repository(name: "FastHub") {
      issue(number: 2224) {
        timeline(first: 30) {
          totalCount
          nodes {
            __typename
            ... on Commit {
              author {
                name
                avatarUrl
              }
              message
            }
            ... on IssueComment {
              author {
                login
                avatarUrl
              }
              createdAt
              authorAssociation
              bodyHTML
              body
              id
              updatedAt
              viewerCanReact
              viewerCanDelete
              viewerCanUpdate
              viewerDidAuthor
              viewerCanMinimize
              viewerCannotUpdateReasons
              reactionGroups {
                content
                createdAt
                viewerHasReacted
              }
            }
            ... on CrossReferencedEvent {
              createdAt
              isCrossRepository
              referencedAt
              source {
                __typename
              }
              willCloseTarget
              actor {
                login
                avatarUrl
              }
            }
            ... on ClosedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              closer {
                ... on Commit {
                  abbreviatedOid
                  author {
                    name
                    avatarUrl
                  }
                  message
                }
                ... on PullRequest {
                  number
                  author {
                    avatarUrl
                    login
                  }
                  merged
                }
              }
            }
            ... on ReopenedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
            }
            ... on SubscribedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
            }
            ... on UnsubscribedEvent {
              actor {
                avatarUrl
                login
              }
              createdAt
            }
            ... on ReferencedEvent {
              actor {
                login
                avatarUrl
              }
              createdAt
              commitRepository {
                nameWithOwner
              }
              isCrossRepository
              isDirectReference
              commit {
                message
                committedDate
                abbreviatedOid
              }
              subject {
                ... on Issue {
                  number
                  author {
                    login
                    avatarUrl
                  }
                  repository {
                    nameWithOwner
                  }
                }
                ... on PullRequest {
                  number
                  author {
                    login
                    avatarUrl
                  }
                  repository {
                    nameWithOwner
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>I really would love to find a quick solution is this block me right now! </p>
<p>Thanks for your help!</p>
<p>---- Edit: </p>
<p>So I thought I might be doing something wrong but I found that the more I query nested objects the response will also return “” can confirm by the below query: </p>
<pre><code>{
  user(login: "k0shk0sh") {
    repository(name: "FastHub") {
      issue(number: 2220) {
        timeline(first: 30) {
          totalCount
          nodes {
            __typename
            ... on Commit {
              abbreviatedOid
              author {
                name
                avatarUrl
              }
              additions
              deletions
              changedFiles
              committedDate
              status {
                state
              }
              oid
            }
            ... on IssueComment {
              id
              databaseId
              author {
                avatarUrl
                login
                url
              }
              authorAssociation
              bodyHTML
              body
              createdAt
              createdViaEmail
              includesCreatedEdit
              isMinimized
              lastEditedAt
              minimizedReason
              publishedAt
              reactionGroups {
                users(first: 0) {
                  totalCount
                }
                createdAt
                viewerHasReacted
                content
                createdAt
              }
              updatedAt
              viewerDidAuthor
              viewerCannotUpdateReasons
              viewerCanUpdate
              viewerCanReact
              viewerCanMinimize
              viewerCanDelete
              userContentEdits(first: 0) {
                totalCount
              }
              url
            }
            ... on CrossReferencedEvent {
              id
              actor {
                login
                avatarUrl
              }
              createdAt
              isCrossRepository
              referencedAt
              url
              willCloseTarget
            }
            ... on ClosedEvent {
              id
              actor {
                avatarUrl
                login
                url
              }
              closable {
                closed
                closedAt
              }
              createdAt
              url
            }
            ... on ReopenedEvent {
              id
              actor {
                avatarUrl
                login
                url
              }
              closable {
                closed
                closedAt
              }
              createdAt
            }
            ... on SubscribedEvent {
              id
              actor {
                avatarUrl
                login
                url
              }
              createdAt
            }
            ... on UnsubscribedEvent {
              id
              createdAt
            }
            ... on ReferencedEvent {
              id
            }
            ... on AssignedEvent {
              id
            }
            ... on UnassignedEvent {
              id
            }
            ... on LabeledEvent {
              id
            }
            ... on UnlabeledEvent {
              id
            }
            ... on MilestonedEvent {
              id
            }
            ... on DemilestonedEvent {
              id
            }
            ... on RenamedTitleEvent {
              id
            }
            ... on LockedEvent {
              id
            }
            ... on UnlockedEvent {
              id
            }
            ... on TransferredEvent {
              id
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>add actor object to </p>
<pre><code>UnsubscribedEvent
</code></pre>
<p>and then the response will always be empty  or any data to any of the empty nodes! </p>

**Comment by nickvanw** [comment, marked as answer]
<p>Hey <a class="mention" href="/k0shk0sh">@k0shk0sh</a>!</p>
<p>It looks like this is a specific problem with the GraphQL Explorer, and not the GraphQL API. I’m not sure why that is at this time (this is the first report I’ve seen of the bug), but I believe you can quickly unblock yourself by using an external GraphQL Client to query the API.</p>
<p>Using the Electron GraphiQL App which can be found here: <a href="https://electronjs.org/apps/graphiql" rel="nofollow noopener">https://electronjs.org/apps/graphiql</a> I was able to generate a Personal Access Token and run your larger query against the API without issue, getting a full response. </p>
<p>You should be able to use Postman or even cURL as well, depending on what you feel comfortable with.</p>
<p>Let me know if that works, and I’ll take a look at seeing why the GraphQL Explorer doesn’t work with those larger queries!</p>
